ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -162,9 +259,13 @@ static void bc_standard_mul(bc_num n1, size_t n1len, bc_num n2, size_t n2len, bc
 	char *pend = pptr + prodlen - 1;
 	i = 0;
 	while (i < prod_arr_size - 1) {
-		for (size_t j = 0; j < BC_MUL_UINT_DIGITS; j++) {
-			*pend-- = prod_uint[i] % BASE;
-			prod_uint[i] /= BASE;
+		if (BC_MUL_UINT_DIGITS == 4) {
+			bc_write_bcd_representation(prod_uint[i], pend - 3);
+			pend -= 4;
+		} else {
+			bc_write_bcd_representation(prod_uint[i] / 10000, pend - 7);
+			bc_write_bcd_representation(prod_uint[i] % 10000, pend - 3);
+			pend -= 8;
 		}
```

Comment:
Maybe?
(By any chance, do compiler optimizations interpret these codes the same way? I'm not very familiar with optimization, so I'm not sure.)

```suggestion
#if BC_MUL_UINT_DIGITS == 4
			bc_write_bcd_representation(prod_uint[i], pend - 3);
			pend -= 4;
#else
			bc_write_bcd_representation(prod_uint[i] / 10000, pend - 7);
			bc_write_bcd_representation(prod_uint[i] % 10000, pend - 3);
			pend -= 8;
#endif
```

Reply:
This should give the same code, but I agree that ifdefs are better here